lua/legendary/init.lua

```diff
@@ -1,141 +1,164 @@
-local _tl_compat; if (tonumber((_VERSION or ''):match('[%d.]*$')) or 0) < 5.3 then local p, m = pcall(require, 'compat53.module'); if p then _tl_compat = m end end; local string = _tl_compat and _tl_compat.string or string
-
-require('legendary.types')
-
-local M = {}
-
-local wk = require('legendary.compat.which-key')
-
-
-
-
-
-
-M.bind_whichkey = wk.bind_whichkey
-
-
-
-
-
-
-
-
-M.parse_whichkey = wk.parse_whichkey
-
-
-
-
-M.whichkey_listen = wk.whichkey_listen
-
-
-local b = require('legendary.bindings')
-
-
-
-
-M.bind_keymap = b.bind_keymap
-
-
-
-
-M.bind_keymaps = b.bind_keymaps
-
-
-
-
-M.bind_command = b.bind_command
-
-
-
-
-M.bind_commands = b.bind_commands
-
-
-
-
-M.bind_autocmds = b.bind_autocmds
-
-
-
-
-M.bind_function = b.bind_function
-
-
-
-
-M.bind_functions = b.bind_functions
-
-
-
-
-
-
-
-
-
-M.find = b.find
+local Config = require('legendary.config')
+local State = require('legendary.data.state')
+local Ui = require('legendary.ui')
+local Executor = require('legendary.api.executor')
+local Keymap = require('legendary.data.keymap')
+local Command = require('legendary.data.command')
+local Augroup = require('legendary.data.augroup')
+local Autocmd = require('legendary.data.autocmd')
+local Function = require('legendary.data.function')
+local LegendaryWhichKey = require('legendary.integrations.which-key')
+
+-- TODO leave this in here for a bit, take it out a few weeks/months
+-- after v2 is launched.
+local function show_notice()
+  vim.defer_fn(function()
+    vim.notify(
+      'legendary.nvim now uses semantic versioning via git tags. If you are experiencing errors with the new version, '
+        .. 'please pin to a version tag or update your configuration. See updated documentation at '
+        .. 'https://github.com/mrjones2014/legendary.nvim for more details.'
+    )
+  end, 500)
+end
 
+if vim.v.vim_did_enter then
+  show_notice()
+else
+  vim.api.nvim_create_autocmd('VimEnter', { callback = show_notice, once = true })
+end
```

Comment:
Also can you think of a better way to do this? Ideally it should be something you can acknowledge and dismiss permanently.

Reply:
Aight this shit is annoying me already. I gotta find a better solution for this. Just going to remove it for now.